File: C/selectionsort/qwen-qwq-32b.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void selectionSort(int arr[], int n) {
    for (int i = 0; i < n-1; i++) {
        int min_idx = i;
        for (int j = i+1; j < n; j++) {
            if (arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        if (min_idx != i) {
            // swap
            int temp = arr[i];
            arr[i] = arr[min_idx];
            arr[min_idx] = temp;
        }
    }
}

int main() {
    FILE *file = fopen("../../inputs.txt", "r");
    if (!file) {
        return 1;
    }
    int arr[1000];
    int n = 0;

    while (fscanf(file, "%d", &arr[n]) == 1 && n < 1000) {
        n++;
    }

    selectionSort(arr, n);
    fclose(file);

    return 0;
}

File: Python/selectionsort/qwen-qwq-32b.py

def selection_sort(arr):
    n = len(arr)
    for i in range(n):
        min_val = arr[i]
        min_idx = i
        for j in range(i+1, n):
            current = arr[j]
            if current < min_val:
                min_val = current
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

import sys

def main():
    input_path = "inputs.txt" if len(sys.argv) < 2 else sys.argv[1]

    try:
        with open(input_path, 'r') as f:
            arr = [int(line.strip()) for line in f if line.strip()]
    except FileNotFoundError:
        return

    sorted_arr = selection_sort(arr)
    

if __name__ == "__main__":
    main()

